SlowCheetah.JDT/Processors/JdtRemove.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Sayed Ibrahim Hashimi. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See  License.md file in the project root for full license information.
+
+namespace SlowCheetah.JDT
+{
+    using System.Linq;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Represents a recursive JDT transformation
+    /// </summary>
+    internal class JdtRemove : JdtProcessor
+    {
+        private const string PathAttribute = "path";
+
+        private bool removedThisNode;
+
+        /// <inheritdoc/>
+        public override string Verb { get; } = "remove";
+
+        /// <inheritdoc/>
+        public override void Process(JObject source, JObject transform)
+        {
+            this.removedThisNode = false;
+
+            JToken removeValue;
+            if (transform.TryGetValue(JsonUtilities.JdtSyntaxPrefix + this.Verb, out removeValue))
+            {
+                this.Remove(source, removeValue, true);
+            }
+
+            if (!this.removedThisNode)
+            {
+                // If the current node was removed, then do not perform any more transformations here
+                this.Successor.Process(source, transform);
+            }
+        }
+
+        private void Remove(JObject source, JToken removeValue, bool allowArray)
```

Comment:
The `allowArray` special case appears to be common among a few processors. Consider extracting to a base class to consolidate the iteration when the transform argument is an array. If it is possible that is. Other issues may prevent that.